File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/coupon/exception/AlreadyIssuedCouponException.java
package band.gosrock.domain.domains.coupon.exception;


import band.gosrock.common.exception.DuDoongCodeException;

public class AlreadyIssuedCouponException extends DuDoongCodeException {
    public static final DuDoongCodeException EXCEPTION = new AlreadyIssuedCouponException();

    private AlreadyIssuedCouponException() {
        super(CouponErrorCode.ALREADY_ISSUED_COUPON);
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/coupon/exception/AlreadyIssuedCouponException.kt
package band.gosrock.domain.domains.coupon.exception

import band.gosrock.common.exception.DuDoongCodeException

class AlreadyIssuedCouponException private constructor() : DuDoongCodeException(CouponErrorCode.ALREADY_ISSUED_COUPON) {
    companion object {
        @JvmField
        val EXCEPTION: DuDoongCodeException = AlreadyIssuedCouponException()
    }
}
